source/host_resolver.c

```diff
@@ -113,6 +150,32 @@ struct default_host_resolver {
     uint32_t pending_host_entry_shutdown_completion_callbacks;
 };
 
+/* Default host resolver implementation for listener. */
+struct host_listener {
+
+    /* Reference to the host resolver that owns this listener */
+    struct aws_host_resolver *resolver;
+
+    /* String copy of the host name */
+    struct aws_string *host_name;
+
+    /* User-supplied callbacks/user_data */
+    aws_host_listener_resolved_address_fn *resolved_address_callback;
+    aws_host_listener_shutdown_fn *shutdown_callback;
+    void *user_data;
+
+    /* Synchronous data, requires host resolver lock to read/modify*/
```

Comment:
Let's put this in its own struct like we usually do, `synced_data` most likely

Reply:
The rest of the host resolver doesn't use that paradigm so I was worried it might be confusing to mix that in.  Might be okay though!

Reply:
We should move toward it over time. For this one, I'd rather break consistency and use the right pattern.